### DISCUSSION TITLE
Controls with the Contact Action

### DISCUSSION THREAD

**Discussion by ABallisat** [idaholab/moose#16293] [Q&A Modules: General] [answered]
Hello all,

I am using the Contact Action to define a mechanical contact and after a period of time I want to turn this off. I am trying this with the Controls system but it appears that the Action cannot be enabled/disabled, giving the following error:
```
*** ERROR ***
The desired parameter 'Contact::leftright/enable' was not located for the 'u_threshold' object, it either does not exist or has not been declared as controllable.
```
I understand that the action sets up various other objects so is there a way to either turn all these off in one command (based on this is seems unlikely https://groups.google.com/g/moose-users/c/jbW2TlY96pE/m/GC2K-GwnCwAJ) or get a list of everything it sets up so I can go through and turn them off? Or any other way of doing it?

If it helps this is my contact block definition:

```
[Contact]
  [./leftright]
    secondary = 'main_right'
    primary = 'right_left'
    model = frictionless
    penalty = 1e+6
    normal_smoothing_distance = 0.1
    order = SECOND
  [../]
[]
```

Thanks,

Alex

**Comment by aeslaughter** [comment, marked as answer]
Controls only work with parameters of MooseObjects. So, you are correct that you will not be able control a parameter of an Action. A possible solution would be to add to the "control_tags" parameter of every object that the Contact action creates. This would require modifications to the contact module, but shouldn't be very difficult.

https://github.com/idaholab/moose/blob/next/test/tests/controls/tag_based_naming_access/object_param.i

**Reply by ABallisat** [reply]
Thanks for this. In the end as I only need to switch it off after a certain time and not turn it back on decided to use the restart system to just spin up a second model with it contact off but using the output of the first sim (with contact), for me it was a simpler solution given my time constraints.
